app/routes/api.circulars.ts

```diff
@@ -192,12 +216,26 @@ export async function action({ request }: ActionFunctionArgs) {
   )
     throw new Response(null, { status: 400 })
 
-  const circular = {
-    submittedHow: 'api',
-    subject,
-    body,
-    ...(format ? { format } : {}),
-  } as const
-
-  return await put(circular, user)
+  switch (request.method) {
+    case 'POST':
+      const circular = {
+        submittedHow: 'api',
+        subject,
+        body,
+        eventId,
+        ...(format ? { format } : {}),
+      } as const
+
+      return await put(circular, user)
+
+    case 'PUT':
+      const circularVersion = {
+        circularId,
+        subject,
+        body,
+        eventId,
+      } as Circular
```

Comment:
Remove the type assertion. You may need to add some runtime checks to clear type checking, but that's a good thing, because it will force you to do the necessary validation.